### ISSUE TITLE
GLEW: OpenGL renderer in Wayland doesn't work.

### ISSUE THREAD

**Issue by DmitryHetman** [SuperTux/supertux#693]
**SuperTux version:**
0.5.1
**System information:** *(details about your kernel, operating system, or distribution)*
Gentoo GNU/Linux nomultilib.
##### Expected behavior
Work with opengl renderer in wayland environment.
##### Actual behavior
segmentation fault
##### Steps to reproduce actual behavior
run supertux2 in wayland environment (only sdl renderer works).
##### Additional debugging information

*If available, please attach a backtrace of a crash (you can get one using GDB or another debugger), or a core dump. If you're on GNU/Linux and your distro is using systemd, it's highly likely that a core dump is available using `coredumpctl(1)`.
If the issue only occurs in a custom level, please attach that too.*

**Comment by tobbi** [comment]
Related: #323. Can you *please* provide a stack backtrace?

**Comment by r-o-a-m** [comment]
<b>supertux version:</b> supertux2 v0.5.0-289-gb96e92ad4
<b>System Information:</b> Debian testing, Intel HD Graphics 520

I'm affected by this issue too. Here's a backtrace from gdb:
```
(gdb) bt
#0  0x00007fb8010d508d in XQueryExtension ()
   from /usr/lib/x86_64-linux-gnu/libX11.so.6
#1  0x00007fb8010c8c22 in XInitExtension ()
   from /usr/lib/x86_64-linux-gnu/libX11.so.6
#2  0x00007fb8030bd7bc in ?? () from /usr/lib/x86_64-linux-gnu/libGL.so.1
#3  0x00007fb8030b8f11 in glXQueryVersion ()
   from /usr/lib/x86_64-linux-gnu/libGL.so.1
#4  0x00007fb802e68d0c in glxewInit ()
   from /usr/lib/x86_64-linux-gnu/libGLEW.so.2.0
#5  0x00007fb802e71053 in glewInit ()
   from /usr/lib/x86_64-linux-gnu/libGLEW.so.2.0
#6  0x000055555580e0bd in GLRenderer::GLRenderer() ()
#7  0x0000555555766dc6 in GLVideoSystem::GLVideoSystem() ()
#8  0x00005555556898b5 in VideoSystem::create(VideoSystem::Enum) ()
#9  0x0000555555659a89 in Main::launch_game() ()
#10 0x000055555565b759 in Main::run(int, char**) ()
#11 0x0000555555656055 in main ()
(gdb) 
```

**Comment by tobbi** [comment]
Looking at the stacktrace, I am inclined to believe that this isn't a SuperTux problem but a problem with glew OR somewhere deeper down.

Are you able to compile SuperTux yourself and use glbinding instead of GLEW and see whether that fixes things?

**Comment by r-o-a-m** [comment]
I recompiled with ```cmake -DGLBINDING_ENABLED=ON .. && make``` and that fixes things. Thank you for pointing me in the right direction!

**Comment by tobbi** [comment]
@nigels-com Can you shed some light onto this issue? Should we create a GitHub issue in the GLEW repository?

**Comment by deveee** [comment]
I didn't look at ST code, but for current GLEW version you can just ignore GLEW_ERROR_NO_GLX_DISPLAY error to make it working with wayland:
https://github.com/supertuxkart/stk-code/commit/2f7d7b677d758755068d0981b3056e5c3e9c8e90#diff-c65927b18aaac2dc90aff9b24bc8c6ae

At least it works in STK. Older GLEW version indeed was causing a crash and you had to patch it to make it working.

**Comment by tobbi** [comment]
This was fixed by @deveee with Pull Request #808. Thanks.
